packages/legend-query/src/stores/QueryBuilderPropertyEditorState.ts

```diff
@@ -178,22 +285,71 @@ export const generateValueSpecificationForParameter = (
   );
 };
 
-const fillDerivedPropertyArguments = (
+export const fillDerivedPropertyArguments = (
   derivedPropertyExpressionState: QueryBuilderDerivedPropertyExpressionState,
-  queryBuilderState: QueryBuilderState,
+  graph: PureModel,
 ): void => {
   const propertyArguments: ValueSpecification[] =
     derivedPropertyExpressionState.parameterValues;
+  let temporalTarget: MILESTONING_STEREOTYPE | undefined;
+  if (
+    derivedPropertyExpressionState.propertyExpression.func.genericType.value
+      .rawType instanceof Class &&
+    derivedPropertyExpressionState.propertyExpression.func.owner
+      ._generatedMilestonedProperties.length !== 0
+  ) {
+    temporalTarget = getMilestoneTemporalStereotype(
+      derivedPropertyExpressionState.propertyExpression.func.genericType.value
+        .rawType,
+      derivedPropertyExpressionState.queryBuilderState.graphManagerState.graph,
+    );
+  }
   derivedPropertyExpressionState.parameters.forEach((parameter, idx) => {
     if (idx < derivedPropertyExpressionState.parameterValues.length) {
+      if (temporalTarget) {
+        const milestoningParameter = fillMilestonedDerivedPropertyArguments(
```

Comment:
Let's cleanup all of these milestoning-related logic and put them in `fillMilestonedDerivedPropertyArguments`. My thinking here is given a derived property parameter, we produce a value, so I wold expect something like


```ts
const parameterValue = generateMilestonedPropertyParameterValue(...)
  ?? enerateValueSpecificationForParameter(
      parameter,
        derivedPropertyExpressionState.queryBuilderState.graphManagerState
          .graph,
    );
```